tests/raftstore/test_transfer_leader.rs

```diff
@@ -138,10 +141,18 @@ fn test_transfer_leader_during_snapshot<T: Simulator>(cluster: &mut Cluster<T>)
         cluster.must_put(key.as_bytes(), value.as_bytes());
     }
 
+    // hook transport and drop all snapshot packet, so follower's status
+    // will stay at snapshot.
+    cluster.hook_transport(DropSnapshot);
     pd_client.must_add_peer(r1, new_peer(3, 3));
+    // a just added peer need wait a couple of ticks, it'll commucation with leader
+    // before getting snapshot
+    sleep_ms(1000);
 
     cluster.transfer_leader(r1, new_peer(3, 3));
```

Comment:
I think you should check transferring to peer 2, and then to check whether the leader is still peer 1.